scripts/darwin.sh

```diff
@@ -1,35 +1,38 @@
-#!/bin/sh
+#!/bin/bash
 
-function get_key {
-  echo $(grep "$1" | awk -F "  +" '{ print $3 }')
+set -u
+set -e
+
+get_key() {
+  grep "$1" | awk -F "  +" '{ print $3 }'
 }
 
-function get_until_paren {
+get_until_paren() {
   awk 'match($0, "\\(|$"){ print substr($0, 0, RSTART - 1) }'
 }
 
-DISKS="`diskutil list | grep '^\/' | get_until_paren`"
-mount_output="`mount`"
+DISKS="$(diskutil list | grep '^\/' | get_until_paren)"
+mount_output="$(mount)"
 
 for disk in $DISKS; do
-  diskinfo="`diskutil info $disk`"
+  diskinfo="$(diskutil info "$disk")"
 
-  device=`echo "$diskinfo" | get_key "Device Node"`
+  device="$(echo "$diskinfo" | get_key "Device Node")"
 
   # See http://superuser.com/q/631592
-  raw_device=`echo "$device" | sed "s/disk/rdisk/g"`
+  raw_device="${device//disk/rdisk}"
 
-  description=`echo "$diskinfo" | get_key "Device / Media Name"`
-  volume_name=`echo "$diskinfo" | get_key "Volume Name"`
-  removable=`echo "$diskinfo" | get_key "Removable Media"`
-  protected=`echo "$diskinfo" | get_key "Read-Only Media"`
-  location=`echo "$diskinfo" | get_key "Device Location"`
-  size=`echo "$diskinfo" | sed 's/Disk Size/Total Size/g' | get_key "Total Size" | perl -n -e'/\((\d+)\sBytes\)/ && print $1'`
+  description="$(echo "$diskinfo" | get_key "Device / Media Name")"
+  volume_name="$(echo "$diskinfo" | get_key "Volume Name")"
+  removable="$(echo "$diskinfo" | get_key "Removable Media")"
+  protected="$(echo "$diskinfo" | get_key "Read-Only Media")"
+  location="$(echo "$diskinfo" | get_key "Device Location")"
+  size="$(echo "$diskinfo" | sed 's/Disk Size/Total Size/g' | get_key "Total Size" | perl -n -e'/\((\d+)\sBytes\)/ && print $1')"
 
-  mountpoints=`echo "$mount_output" | perl -n -e'm{^'"${disk}"'(s[0-9]+)? on (.*) \(.*\)$} && print "$2\n"'`
+  mountpoints="$(echo "$mount_output" | perl -n -e'm{^'"${disk}"'(s[0-9]+)? on (.*) \(.*\)$} && print "$2\n"')"
 
   # Omit mounted DMG images
-  if [ "$description" == "Disk Image" ]; then
+  if [ "$description" = "Disk Image" ]; then
```

Comment:
True. This was caused due to the shell change confusion. I left all of them as `==`.